### PR TITLE
Adding a disabling button feature with new css style

### PR DESCRIPTION
**Pull request by PabloFLPs**
Just adding a feature to disable button on form submitting (and also, enabling it again if the promise results in an error). Also, I needed the previous PR, but as it have only a few changes, I opted to add them here and make another PR; this way, I would not need you to approve the other one to make these changes.

### CODE REVIEW

**Comment by PabloFLPs** [comment]
Oh, other thing: I preferred to keep the button disabled after the form submitting (if it is succesful) and not only while tht Heroku is receiving the data. But if you prefer the way you sugested, I can change any time.

File: src/components/Contact.tsx [hunk 1]
```diff
@@ -46,10 +48,15 @@ export default function Contact() {
...
-    .then(successToastEmitter).catch(error => {
+    setDisableButtonState(true)
+
+    try {
+      await axios.post("https://email-free-api.herokuapp.com/send-email", formToBeSubmitted)
+      successToastEmitter()
+    } catch (error) {
       if(error) errorToastEmitter()
-    })
+      setDisableButtonState(false)
```
Comment (Erickinhou) [inline src/components/Contact.tsx]: The setDisableButtonState needs to be out of the `try/catch` statement.

File: src/styles/modules/_contact.module.scss [hunk 2]
```diff
@@ -9,11 +9,17 @@
...
   }
 
   button:hover {
-    filter: brightness(0.8);
+    filter: brightness(0.8) opacity(1);
   }
-}
\ No newline at end of file
+  
+  button:disabled {
```
Comment (Erickinhou) [inline src/styles/modules/_contact.module.scss:21]: Cool

**Comment by Erickinhou** [comment]
> Oh, other thing: I preferred to keep the button disabled after the form submitting (if it is succesful) and not only while tht Heroku is receiving the data. But if you prefer the way you sugested, I can change any time.

If the user accidentally sends the message, does he need to reload the page?

**Review by Erickinhou — CHANGES_REQUESTED** [review]
I hit the submit button and the button is still disabled. If the user accidentally sends the message, does he need to reload the page? I don't think this is a good idea.
image

File: src/components/Contact.tsx (hunk 1 above)
Comment (PabloFLPs) [reply src/components/Contact.tsx]: Well, I don't like this way, but ok. I will makes the changes here.

**Review by Erickinhou — APPROVED** [review]

**Pull request merged**